internal/zinc-core/src/main/scala/sbt/internal/inc/MiniSetupUtil.scala

```diff
@@ -127,4 +126,26 @@ object MiniSetupUtil {
       def equiv(a: CompileOrder, b: CompileOrder) = a == b
     }
   }
+
+  def equivScalacOptions(ignoredRegexes: Array[String]): Equiv[Array[String]] = {
+    def groupWithParams(opts: Array[String]): Set[String] = {
+      def isParam(s: String) = !s.startsWith("-")
+      def recur(opts: List[String], res: Set[String]): Set[String] = opts match {
+        case opt :: param :: rest if isParam(param) => recur(rest, res + (opt + " " + param))
```

Comment:
Maybe string interpolation? It will be more readable (especially that we are using `+` to add elements to Set